client/components/add-record.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { Form, Button, Row, Col, InputGroup, FormControl } from 'react-bootstrap';
+class AddForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      missingInput: false,
+      date: null,
+      cost: null,
+      record: null,
+      mileage: null
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.reset = this.reset.bind(this);
+  }
+
+  handleSubmit(event) {
+    if (event) {
+      event.preventDefault();
+    }
+
+    const { date, cost, record, mileage } = this.state;
+    if (!date || !cost || !record || !mileage) {
+      this.setState({
+        missingInput: true
+      });
+      return;
+    }
+
+    const newRecord = { date, cost, record, mileage };
+    fetch(`/api/garage/add-record/${this.props.vehicleId}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newRecord)
+    })
+      .then(result => result.json())
+      .then(data => {
+        this.reset();
+        this.props.toggleModal();
+        this.props.getHistory();
```

Comment:
instead of re-fetching the whole history, try adding it to the existing history of the parent by sending `data` "up".

```js
this.props.addRecord(data);
```

Reply:
entry now gets sent back to parent to add to records
![image](https://user-images.githubusercontent.com/48267398/163653259-9a06b58c-a7fb-4caf-b65b-f3f94a94ed06.png)